src/java.base/share/classes/java/io/ObjectStreamClass.java

```diff
@@ -830,7 +830,7 @@ void readNonProxy(ObjectInputStream in)
             char tcode = (char) in.readByte();
             String fname = in.readUTF();
             String signature = ((tcode == 'L') || (tcode == '[')) ?
-                in.readTypeString() : new String(new char[] { tcode });
+                in.readTypeString() : String.valueOf(tcode);
```

Comment:
Certainly more readable and it seems that the call to valueOf is equivalent to whay takes place with the original code. I can't see any difference semantically or performance-wise at a glance. LGTM

Reply:
@c-cleary Thanks for review. The difference is about intermediate array: `String.valueOf()` doesn't allocate it being slightly faster and less memory-consuming.

Could you file an issue to track this? I'm not an Oracle employee and not able to do it myself.

Reply:
Nice cleanup. I can help file a JBS issue if @c-cleary doesn't notice your comment.

Reply:
@kelthuzadx hi! I'd appreciate this, as there's no JBS issue for this (

Reply:
Hi @stsypanov, I've created it https://bugs.openjdk.java.net/browse/JDK-8263552. Good luck :-)

Reply:
Thanks!